07_errors/hsy3418/store_test.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+func (suite *storerTestSuite) SetupTest() {
+	suite.store = suite.makeStore()
+	suite.toBeCreatedPuppy = Puppy{ID: 101, Breed: "Poodle", Colour: "White", Value: 1000.5}
+	suite.existsPuppy = Puppy{ID: 102, Breed: "Poodle", Colour: "White", Value: 1280.5}
+	suite.toBeUpdatedPuppy = Puppy{ID: 102, Breed: "Poodle", Colour: "White", Value: 2000}
+	suite.invalidPuppy = Puppy{ID: 103, Breed: "Poodle", Colour: "White", Value: -1000}
+	suite.invaildError = &Error{
+		Message: "The puppy value is invalidate",
+		Code:    -1,
+	}
+	suite.invalidIDError = &Error{Message: "puppy ID does not exist", Code: -2}
+	err := suite.store.CreatePuppy(suite.existsPuppy)
+	if err != nil {
+		suite.FailNow("Failed to setup test")
+	}
+}
+
+type storerTestSuite struct {
+	suite.Suite
+	store            Storer
+	makeStore        func() Storer
+	toBeCreatedPuppy Puppy
+	existsPuppy      Puppy
+	toBeUpdatedPuppy Puppy
+	invalidPuppy     Puppy
+	invaildError     error
+	invalidIDError   error
+}
+
+//successfully create puppy, new puppy, existing puppy, value <0
+func (suite *storerTestSuite) TestCreatePuppy() {
+	assert := assert.New(suite.T())
+	testCases := []struct {
+		title    string
+		input    Puppy
+		expected error
+	}{
+		{"Create new puppy", suite.toBeCreatedPuppy, nil},
+		{"Create existing puppy", suite.toBeCreatedPuppy, fmt.Errorf("puppy with 101 ID already exists")},
+		{"Create a invalid puppy", suite.invalidPuppy, suite.invaildError},
+	}
+	for _, tc := range testCases {
+		tc := tc
+		suite.T().Run(tc.title, func(t *testing.T) {
+			err := suite.store.CreatePuppy(tc.input)
+			assert.Equal(tc.expected, err)
+		})
+	}
+
```

Comment:
Unnecessary blank line.
No need for blank lines at beginning or end of block.
Use blank lines between type, method and function definitions and within a block for logical separation.  
(below as well)